packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -92,9 +92,10 @@ String runSync(List<String> cmd, { String workingDirectory }) {
 }
 
 /// Return the platform specific name for the given Dart SDK binary. So, `pub`
-/// ==> `pub.bat`.
-String sdkBinaryName(String name) {
-  return path.absolute(path.join(dartSdkPath, 'bin', Platform.isWindows ? '$name.bat' : name));
+/// ==> `pub.bat`.  The default SDK location can be overridden with a specified
+/// [sdkLocation].
+String sdkBinaryName(String name, { String sdkLocation }) {
+  return path.absolute(path.join(sdkLocation ?? dartSdkPath, 'bin', Platform.isWindows ? '$name.bat' : name));
```

Comment:
Re-cap of side conversation.  The rub is that we can't used `defaultsTo:` since when it gets called, `ArtifactStore.flutterRoot` isn't set yet.  Anyway, we agreed to revisit if needed.